DQM/HLTEvF/plugins/HLTObjectMonitorHI.cc

```diff
@@ -0,0 +1,778 @@
+// -*- C++ -*-
+//
+// Package:    QM/HLTObjectMonitorHI
+// Class:      HLTObjectMonitorHI
+//
+/**\class HLTObjectMonitorHI HLTObjectMonitorHI.cc DQM/HLTEvF/plugins/HLTObjectMonitorHI.cc 
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Geonhee Oh
+//         Created:  Sun, 29 Oct 2018 17:00:00 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+#include <sys/time.h>
+#include <cstdlib>
+
+// user include files
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+
+//for collections
+#include "HLTrigger/JetMET/interface/AlphaT.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidate.h"
+#include "DataFormats/BTauReco/interface/JetTag.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidateFwd.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/CaloJet.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+#include "TMath.h"
+#include "TStyle.h"
+#include "TLorentzVector.h"
+
+#include <unordered_map>
+//
+// class declaration
+//
+
+//using namespace edm;
+using namespace trigger;
+using std::vector;
+using std::string;
+using std::unordered_map;
+
+class HLTObjectMonitorHI : public DQMEDAnalyzer {
```

Comment:
if you plan to run this module on the HLT farm, rather than in the online DQM, please inherit from `DQMGlobalEDAnalyzer`

Reply:
OK, this will *not* run as part of the HLT, only in the online DQM.